tests/compile-fail/lint_pass.rs

```diff
@@ -0,0 +1,31 @@
+#![feature(plugin)]
+#![feature(rustc_private)]
+#![plugin(clippy)]
+
+#![deny(lint_without_lint_pass)]
+//~^ NOTE:
+
+#[macro_use] extern crate rustc;
+
+use rustc::lint::{LintPass, LintArray};
+
+declare_lint! { GOOD_LINT, Warn, "good lint" }
+declare_lint! { MISSING_LINT, Warn, "missing lint" }
+//(DISABLED)~^ ERROR: the lint MISSING_LINT is not added to any LintPass</del>
+
+// FIXME: Either the JSON output or compiletest-rs is wrong. The error above is
+//        not captured.
```

Comment:
That's weird. The error is in the JSON:

```json
{"message":"the lint MISSING_LINT is not added to any LintPass","code":null,"level":"error","spans":[{"file_name":"<rustc macros>","byte_start":6188,"byte_end":6293,"line_start":5,"line_end":6,"column_start":1,"column_end":30,"is_primary":true,"text":[{"text":"static $ name : & 'static :: rustc :: lint :: Lint = & lint_initializer ! (","highlight_start":1,"highlight_end":76},{"text":"$ name , $ level , $ desc ) ; ) ;","highlight_start":1,"highlight_end":30}],"label":null,"suggested_replacement":null,"expansion":{"span":{"file_name":"tests/compile-fail/lint_pass.rs","byte_start":234,"byte_end":286,"line_start":13,"line_end":13,"column_start":1,"column_end":53,"is_primary":false,"text":[{"text":"declare_lint! { MISSING_LINT, Warn, \"missing lint\" }","highlight_start":1,"highlight_end":53}],"label":null,"suggested_replacement":null,"expansion":null},"macro_decl_name":"declare_lint!","def_site_span":{"file_name":"<rustc macros>","byte_start":5954,"byte_end":6297,"line_start":1,"line_end":6,"column_start":1,"column_end":34,"is_primary":false,"text":[{"text":"( pub $ name : ident , $ level : ident , $ desc : expr ) => (","highlight_start":1,"highlight_end":62},{"text":"pub static $ name : & 'static :: rustc :: lint :: Lint = & lint_initializer !","highlight_start":1,"highlight_end":78},{"text":"( $ name , $ level , $ desc ) ; ) ; (","highlight_start":1,"highlight_end":38},{"text":"$ name : ident , $ level : ident , $ desc : expr ) => (","highlight_start":1,"highlight_end":56},{"text":"static $ name : & 'static :: rustc :: lint :: Lint = & lint_initializer ! (","highlight_start":1,"highlight_end":76},{"text":"$ name , $ level , $ desc ) ; ) ;","highlight_start":1,"highlight_end":34}],"label":null,"suggested_replacement":null,"expansion":null}}}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"tests/compile-fail/lint_pass.rs","byte_start":75,"byte_end":97,"line_start":5,"line_end":5,"column_start":9,"column_end":31,"is_primary":true,"text":[{"text":"#![deny(lint_without_lint_pass)]","highlight_start":9,"highlight_end":31}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null},{"message":"for further information visit https://github.com/Manishearth/rust-clippy/wiki#lint_without_lint_pass","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":null}
```

but does not seem to be seen by compiletest-rs:
```
actual errors (from JSON output): [
    Error {
        line_num: 5,
        kind: Some(
            Note
        ),
        msg: "5:9: 5:31: lint level defined here"
    }
]

expected errors (from test file): [
    Error {
        line_num: 5,
        kind: Some(
            Note
        ),
        msg: ""
    },
    Error {
        line_num: 13,
        kind: Some(
            Error
        ),
        msg: "the lint MISSING_LINT is not added to any LintPass</del>"
    }
]
```